### PR TITLE
Chore: Remove more ESLintRC-related files (refs #13481)

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

As part of #13481, I've refactored tests to avoid using the in-memory filesystem helper.

#### What changes did you make? (Give an overview)

Changed tests for `CLIEngine`, `ESLint`, and `FileEnumerator` to use an actual filesystem instead of an in-memory one. Also removed `CascadingConfigArrayFactory` and `ConfigArrayFactory` as these are now included from `@eslint/eslintrc`.

I didn't remove the in-memory filesystem utility because it is still in use by some other tests. 

#### Is there anything you'd like reviewers to focus on?

Did I miss any related functionality?

### CODE REVIEW

**Comment by nzakas** [comment]
Apologies, I thought this was ready but then the last step seemed to mess everything up. I'll continue working.

**Comment by mdjermanovic** [comment]
Removed leading and trailing spaces from the title.

**Comment by nzakas** [comment]
I had to adjust the coverage settings across all four dimensions to get this to pass, but I think it's worth it.

File: tests/lib/eslint/eslint.js [hunk 1]
```diff
@@ -3659,46 +3784,71 @@ describe("ESLint", () => {
...
+                await cleanup();
+
+                const configFilePath = path.resolve(root, "../.eslintrc.json");
+
+                if (shell.test("-e", configFilePath)) {
+                    shell.rm(configFilePath);
+                }
+            });
+
+            it("'executeOnFiles(\".\")' should not load config files from outside of \".\".", async () => {
```
Comment (btmills) [inline tests/lib/eslint/eslint.js:3803]: Did you mean to change `lintFiles` to `executeOnFiles`?

```suggestion
            it("'lintFiles(\".\")' should not load config files from outside of \".\".", async () => {
```

**Review by btmills — APPROVED** [review]
I underestimated just how much code extracting eslintrc would remove from core.

**Review by btmills — APPROVED** [review]
LGTM. Merging this into today's release so you can keep working.

**Pull request merged**